internal/servers/destination/v1/destinations.go

```diff
@@ -93,32 +116,40 @@ func (s *Server) Write(msg pb.Destination_WriteServer) error {
 			return status.Errorf(codes.InvalidArgument, "failed to unmarshal source spec: %v", err)
 		}
 	}
-	syncTime := r.Timestamp.AsTime()
 	s.setPKsForTables(tables)
 	eg, ctx := errgroup.WithContext(msg.Context())
+
 	eg.Go(func() error {
-		return s.Plugin.Write(ctx, sourceSpec, tables, syncTime, resources)
+		return s.Plugin.Write(ctx, plugin.WriteOptions{
+			MigrateForce: s.spec.MigrateMode == specs.MigrateModeForced,
+		}, msgs)
 	})
 
+	for _, table := range tables {
```

Comment:
```suggestion
	for _, table := range tables.FlattenTables() {
```
Not sure if tables are unflattened here (or in Migrate) but we seem to be calling MigrateTable a lot?